### PR TITLE
Added unit tests for ecc.c

### PR DESCRIPTION
**Pull request by ethanlooney**

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -20598,8 +20598,75 @@ static int test_wc_ecc_is_valid_idx (void)
 
 
 } /* END test_wc_ecc_is_valid_idx */
+/*
+ * Testing wc_ecc_get_curve_id_from_oid()
+ */
+static int test_wc_ecc_get_curve_id_from_oid (void)
+{
+    int ret = 0;
+#if (!defined(NO_ECC256)  || defined(HAVE_ALL_CURVES))
```
Comment (cconlon) [inline tests/api.c:20607]: Please equalize spacing here after first "!defined()"

File: tests/api.c [hunk 2]
```diff
@@ -20598,8 +20598,75 @@ static int test_wc_ecc_is_valid_idx (void)
...
+{
+    int ret = 0;
+#if defined(HAVE_ECC) && !defined(WC_NO_RNG)
+    ecc_key     key;
+    WC_RNG      rng;
+    int         sz;
+
+    printf(testingFmt, "wc_ecc_sig_size_calc()");
+
+    ret = wc_InitRng(&rng);
```
Comment (cconlon) [inline tests/api.c:20650]: Please free WC_RNG when finished

File: tests/api.c [hunk 3]
```diff
@@ -20598,8 +20598,75 @@ static int test_wc_ecc_is_valid_idx (void)
...
+    if (ret == 0) {
+        ret = wc_ecc_init(&key);
+        if (ret == 0) {
+            ret = wc_ecc_make_key(&rng, 16, &key);
```
Comment (cconlon) [inline tests/api.c:20654]: Please free ecc_key when finished

File: tests/api.c [hunk 4]
```diff
@@ -20598,8 +20598,75 @@ static int test_wc_ecc_is_valid_idx (void)
...
+        }
+    }
+    sz = sizeof(key);
```
Comment (cconlon) [inline tests/api.c:20657]: sz is being set here as the size of the ecc_key structure.  It should represent the size of the underlying key, in bytes.  You may be able to get it from "sz = key.dp->size"

**Comment by ethanlooney** [comment]
Closed due to merge conflicts, opening a new pull request with updated code that fixed the merge conflict.

**Pull request closed without merging**